### PR TITLE
Fix failing build if no error page

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description
This PR fixes a bug which causes the build to fail if an error page has not been created in the CMS. It introduces default values for the error page.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/lib/data/common/index.js [hunk 1]
```diff
@@ -49,6 +49,11 @@ export async function getPageProps(api, context) {
   const {
     docs: [page],
   } = await api.findPage(slug);
+  if (!page && (slug === "404" || slug === "500")) {
+    return {
+      title: "Page not found",
+    };
+  }
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/common/index.js:56]: 1. Don't different error pages have different titles?
2. Don't they also have subtitles/descriptions?

---

PS: Doesn't 500 page have to be static?

Comment (kelvinkipruto) [reply apps/codeforafrica/src/lib/data/common/index.js:56]: @kilemensi The error pages all have the same title and description.

Comment (kilemensi) [reply apps/codeforafrica/src/lib/data/common/index.js]: Can't be @kelvinkipruto: https://github.com/CodeForAfrica/ui/blob/main/apps/codeforafrica/content/pages/error.md

File: apps/codeforafrica/src/lib/data/common/index.js [hunk 2]
```diff
@@ -44,11 +44,113 @@ function getPageSlug({ params }) {
   return params?.slugs?.[pageSlugIndex] || "index";
 }
 
+function getDefaultErrorPageProps(statusCode = 404) {
+  if (statusCode === 500) {
+    return {
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/common/index.js:49]:
```suggestion
function getDefaultErrorPageProps(slug = "404) {
  if (slug === "500") {
    return {
```

Why `slug` to `statusCode` if `slug` can do?

File: apps/codeforafrica/src/lib/data/common/index.js [hunk 3]
```diff
@@ -44,11 +44,113 @@ function getPageSlug({ params }) {
...
   } = await api.findPage(slug);
+  if (!page) {
+    if (slug === "500") {
+      return getDefaultErrorPageProps(500);
+    }
+    return getDefaultErrorPageProps();
+  }
   if (!page) {
     return null;
   }
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/common/index.js:156]:
```suggestion
    if (["404", "500"].includes(slug)) {
      return getDefaultErrorPageProps(slug);
    }
    return null;
  }
```

A single `if (!page)` should be enough.

**Review by kilemensi — APPROVED** [review]

**Pull request merged**